Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/ProximityChecker.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Informs about the proximity staus of the local player
+    /// </summary>
+    public class ProximityChecker : MonoBehaviour
+    {
+        [SerializeField, Tooltip("At which distance will the player be considered 'close'?")]
+        float m_ActivationRadius = 1;
+
+        [SerializeField, Tooltip("A visual representation of the radius?")]
+        Transform m_RadiusRepresentation;
+        Transform m_Transform;
+
+        event Action<bool> OnLocalPlayerProximityStatusChanged;
+        internal bool LocalPlayerIsClose { get; private set; }
+        void Awake()
+        {
+            m_Transform = transform;
+            if (m_RadiusRepresentation)
+            {
+                const float k_OffsetFromGround = 0.01f;
+                m_RadiusRepresentation.transform.localPosition = new Vector3(0, (m_Transform.lossyScale.y / -2) + k_OffsetFromGround, 0);
+            }
+        }
+
+        internal void AddListener(Action<bool> callback)
+        {
+            OnLocalPlayerProximityStatusChanged += callback;
+        }
+
+        internal void RemoveListener(Action<bool> callback)
+        {
+            OnLocalPlayerProximityStatusChanged -= callback;
+        }
```

Comment:
Are these used anywhere? It seems like you're only polling for LocalPlayerIsClose externally, so OnLocalPlayerProximityStatusChanged & the add/remove listener methods are unused essentially.

Reply:
I just added some code that shows how to use them